### PR TITLE
**Kwargs in add_camera

### PR DESCRIPTION
**Pull request by moealani**

### CODE REVIEW

**Comment by moealani** [comment]
Not sure how to fix the error I am getting on CircleCI. I assume it is an easy fix though.

File: pythonAPIClient/client.py [hunk 1]
```diff
@@ -420,43 +420,43 @@ def add_camera(self, camera_type, is_active_image, is_active_video, snapshot_url
...
-                    'video_path': local_params.pop('video_path')
+        if kwargs.get('type') == 'ip':
+            if kwargs.get('ip') is not None:
+                kwargs['retrieval'] = {
+                    'ip': kwargs.pop('ip'),
+                    'port': kwargs.pop('port'),
+                    'brand': kwargs.pop('brand'),
+                    'model': kwargs.pop('model'),
+                    'image_path': kwargs.pop('image_path'),
+                    'video_path': kwargs.pop('video_path')
```
Comment (mush-zhang) [inline pythonAPIClient/client.py]: you cannot guarantee that these keys exists. if user does not provide model, then dict.pop('model') will raise a key error. specifying default value when no key found. like `'model': kwargs.pop('model', None);`

**Review by mush-zhang — CHANGES_REQUESTED** [review]

File: pythonAPIClient/client.py [hunk 2]
```diff
@@ -420,43 +420,43 @@ def add_camera(self, camera_type, is_active_image, is_active_video, snapshot_url
                     The new camera ID for the successfully updated camera.
         """
 
-        local_params = dict(locals())
+        required_args = ('type', 'is_active_image', 'is_active_video',
+                         'snapshot_url', 'm3u8_url', 'ip')
```
Comment (mush-zhang) [inline pythonAPIClient/client.py]: ip, snapshot_url, m3u8_url, and ip should not be required

**Review by mush-zhang — CHANGES_REQUESTED** [review]

File: pythonAPIClient/client.py [hunk 3]
```diff
@@ -420,43 +421,42 @@ def add_camera(self, camera_type, is_active_image, is_active_video, snapshot_url
...
-                local_params['retrieval'] = json.dumps(local_params['retrieval'])
-        elif camera_type == 'non-ip':
-            if snapshot_url is not None:
-                local_params['retrieval'] = {
-                    'snapshot_url': local_params.pop('snapshot_url')
+                kwargs['retrieval'] = json.dumps(kwargs['retrieval'])
+        elif kwargs.get('type') == 'non-ip':
+            if kwargs.get('snapshot_url') is not None:
+                kwargs['retrieval'] = {
+                    'snapshot_url': kwargs.pop('snapshot_url', None)
```
Comment (mush-zhang) [inline pythonAPIClient/client.py]: we can simplify retrieval of by `kwargs['retrieval'] = { 'snapshot_url' : kwargs.pop('snapshot_url', None)` without first checking if it is there. if the any of the fields are not specified, they will be None and error will be returned by the api.
same for ip and stream camera.

**Review by mush-zhang — CHANGES_REQUESTED** [review]

**Review by jaydenC88 — APPROVED** [review]
Good job

**Review by mush-zhang — APPROVED** [review]

**Pull request merged**